### ISSUE TITLE
"Non-readable POM" for non-existent module from inactive profile

### ISSUE THREAD

**Issue by ginkel** [qoomon/maven-git-versioning-extension#106]
Hi there,

let's say, we have a multi-module Maven project where a module is declared as optional based on a Maven profile that is inactive by default:

```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>de.tgbyte.reproducer</groupId>
    <artifactId>maven-git-versioning-extension-reproducer-parent</artifactId>
    <version>0.0.1-SNAPSHOT</version>
    <packaging>pom</packaging>

    <modules>
    </modules>

    <profiles>
       <profile>
            <id>optional-module</id>
            <activation>
                <activeByDefault>false</activeByDefault>
                <file>
                    <exists>${maven.multiModuleProjectDirectory}/optional-module/pom.xml</exists>
                </file>
            </activation>
            <modules>
                <module>optional-module</module>
            </modules>
        </profile>
    </profiles>
</project>
```

This POM builds successfully (when optional-module is absent from the file system) unless the `maven-git-versioning-extension` is active, which yields the following error:

```
[ERROR]   The project  (/home/tg/workspace/maven-git-versioning-extension-reproducer/pom.xml) has 1 error
[ERROR]     Non-readable POM /home/tg/workspace/maven-git-versioning-extension-reproducer/pom.xml: /home/tg/workspace/maven-git-versioning-extension-reproducer/optional-module (No such file or directory)
java.io.FileNotFoundException: /home/tg/workspace/maven-git-versioning-extension-reproducer/optional-module (No such file or directory)
    at java.io.FileInputStream.open0 (Native Method)
    at java.io.FileInputStream.open (FileInputStream.java:211)
    at java.io.FileInputStream.<init> (FileInputStream.java:153)
    at me.qoomon.maven.gitversioning.MavenUtil.readModel (MavenUtil.java:30)
    at me.qoomon.maven.gitversioning.GitVersioningModelProcessor.determineRelatedProjects (GitVersioningModelProcessor.java:742)
    at me.qoomon.maven.gitversioning.GitVersioningModelProcessor.determineRelatedProjects (GitVersioningModelProcessor.java:708)
    at me.qoomon.maven.gitversioning.GitVersioningModelProcessor.init (GitVersioningModelProcessor.java:189)
    at me.qoomon.maven.gitversioning.GitVersioningModelProcessor.processModel (GitVersioningModelProcessor.java:215)
    at me.qoomon.maven.gitversioning.GitVersioningModelProcessor.read (GitVersioningModelProcessor.java:118)
    at org.apache.maven.model.building.DefaultModelBuilder.readModel (DefaultModelBuilder.java:552)
    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:281)
    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:448)
    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:414)
    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:377)
    at org.apache.maven.graph.DefaultGraphBuilder.collectProjects (DefaultGraphBuilder.java:414)
    at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor (DefaultGraphBuilder.java:405)
    at org.apache.maven.graph.DefaultGraphBuilder.build (DefaultGraphBuilder.java:82)
    at org.apache.maven.DefaultMaven.buildGraph (DefaultMaven.java:507)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:219)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:64)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

A full reproducer is available at: https://github.com/tgbyte/maven-git-versioning-extension-reproducer

Thanks,
Thilo

**Comment by qoomon** [comment]
Hey Thilo, thanks for reporting. I'll add a check to skip non existing files. However I don't get your use case :-). You can expect the fix release this week

**Comment by qoomon** [comment]
Should be fixed in version `6.4.3`, it may take some minutes to get available in maven central.

**Comment by ginkel** [comment]
Excellent, thanks a lot! Tested successfully.
